rfd/0XYZ-nested-accesslists.md

```diff
@@ -0,0 +1,121 @@
+---
+title: RFD 0XYZ - Nested Access lists
+authors: Alex McGrath (alex.mcgrath@goteleport.com)
+state: in development
+---
+
+# Related
+  * Partial implementation PR: https://github.com/gravitational/teleport/pull/38738
+
+# What
+
+This RFD proposes adding support for including owners and
+members in access lists from other access lists
+
+# Why
+
+This allows a hierarchical access list structure for organizations
+which is supported by access list implementations being integrated
+with, for example Azure Entra supports including groups in other
+groups which we want to be able to mimic the structure of in Teleport
+access lists.
+
+Users in an access list hierarchy will inherit the granted roles and
+traits for members and owners from access lists referencing the lists
+they're in.
+
+# Implementations
+
+New fields will be introduced into the access_list type:
+
+```yaml
+kind: access_list
+metadata:
+  name: e69aa529-2a7f-4de2-87be-97fd94309a9f
+spec:
+  grants:
+    roles:
+    - access
+    - auditor
+    traits: {}
+  # list of references to other access lists, for users to include in this access list
+  member_access_lists:
```

Comment:
Maybe this is a basic question, but why do we need member and owners list separate? I'd assume that an access list has "child" access lists and the owners and members are inherited without modifications.

Reply:
For ownership of ALs imported from Entra, it is a plan to have something simple for now, e.g. a static list of owners `["alice", "bob"]` for all imported access lists, so for Entra the `owner_access_lists` is not particularly relevant.

Reply:
I think it makes sens to only have the one way to include access lists 👍

Reply:
@lxea I think we keep separate access lists for owners and members like you have now. Otherwise, how do I set `IT-Team` access list as an owner, and `Dev-Team` as a member?

Reply:
That looks like a good system to me 👍

Reply:
Forgot to reply here earlier but Marek's proposed structure looks good to me too.